c++/src/CMakeLists.txt

```diff
@@ -184,7 +184,11 @@ set(SOURCE_FILES
   Timezone.cc
   TypeImpl.cc
   Vector.cc
-  Writer.cc)
+  Writer.cc
+  CpuInfoUtil.cc
+  BpackingDefault.cc
+  BpackingAvx512.cc
+  Bpacking.cc)
```

Comment:
Can we append these files to `SOURCE_FILES` only when `BUILD_ENABLE_AVX512` is true?
Then we probably do not need to add many macros to disable compilation in the source files.

Reply:
OK, already changed.

Reply:
https://github.com/wpleonardo/orc/blob/fe5b6c7a29721bb5d8c4699a0b072d64555d600d/c%2B%2B/src/CMakeLists.txt#L196
Already changed the building source_files when Build_Enable_AVX512 is true. And also changed the function definitions about  BitUnpack::readLongs, BitUnpackDefault::readLongs, and BitUnpackAVX512::readLongs